engine/src/main/java/io/zeebe/engine/nwe/event/IntermediateCatchEventProcessor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.nwe.event;
+
+import io.zeebe.engine.nwe.BpmnElementContext;
+import io.zeebe.engine.nwe.BpmnElementProcessor;
+import io.zeebe.engine.nwe.behavior.BpmnBehaviors;
+import io.zeebe.engine.nwe.behavior.BpmnEventSubscriptionBehavior;
+import io.zeebe.engine.nwe.behavior.BpmnIncidentBehavior;
+import io.zeebe.engine.nwe.behavior.BpmnStateBehavior;
+import io.zeebe.engine.nwe.behavior.BpmnStateTransitionBehavior;
+import io.zeebe.engine.nwe.behavior.BpmnVariableMappingBehavior;
+import io.zeebe.engine.processor.workflow.deployment.model.element.ExecutableCatchEventElement;
+
+public class IntermediateCatchEventProcessor
+    implements BpmnElementProcessor<ExecutableCatchEventElement> {
+
+  private final BpmnEventSubscriptionBehavior eventSubscriptionBehavior;
+  private final BpmnIncidentBehavior incidentBehavior;
+  private final BpmnStateBehavior stateBehavior;
+  private final BpmnStateTransitionBehavior stateTransitionBehavior;
+  private final BpmnVariableMappingBehavior variableMappingBehavior;
+
+  public IntermediateCatchEventProcessor(final BpmnBehaviors bpmnBehaviors) {
+    eventSubscriptionBehavior = bpmnBehaviors.eventSubscriptionBehavior();
+    incidentBehavior = bpmnBehaviors.incidentBehavior();
+    stateBehavior = bpmnBehaviors.stateBehavior();
+    stateTransitionBehavior = bpmnBehaviors.stateTransitionBehavior();
+    variableMappingBehavior = bpmnBehaviors.variableMappingBehavior();
+  }
+
+  @Override
+  public Class<ExecutableCatchEventElement> getType() {
+    return ExecutableCatchEventElement.class;
+  }
+
+  @Override
+  public void onActivating(
+      final ExecutableCatchEventElement element, final BpmnElementContext context) {
+    variableMappingBehavior
+        .applyInputMappings(context, element)
+        .flatMap(ok -> eventSubscriptionBehavior.subscribeToEvents(element, context))
+        .ifRightOrLeft(
+            ok -> stateTransitionBehavior.transitionToActivated(context),
+            failure -> incidentBehavior.createIncident(failure, context));
+  }
+
+  @Override
+  public void onActivated(
+      final ExecutableCatchEventElement element, final BpmnElementContext context) {
+    if (element.isNone()) {
+      stateTransitionBehavior.transitionToCompleting(context);
+    }
+  }
```

Comment:
An intermediate catch event must have an event definition. I don't see a case when it has no event definition and we complete it immediately. I know it comes from the existing code but I have no idea why :sweat_smile: 
Let's remove it if you agree.

Reply:
Ah, I was unaware we didn't allow these. I'd argue that there are use cases for a none intermediate catch event (e.g. milestone events). But allowing those is out of the scope of these changes. Should I create a new issue for this or is there a clear reason why we don't allow none intermediate catch events? I was unable to find GH issues about this topic.

Reply:
I guess you mean the none **throw** event :stuck_out_tongue_winking_eye: 

I see no reason why we should not implement it. I guess that nobody asked for it yet.